src/libraries/Common/src/System/Drawing/KnownColorTable.cs

```diff
@@ -7,176 +7,252 @@ namespace System.Drawing
-            0xFFFFB6C1,     // LightPink
-            0xFFFFA07A,     // LightSalmon
-            0xFF20B2AA,     // LightSeaGreen
-            0xFF87CEFA,     // LightSkyBlue
-            0xFF778899,     // LightSlateGray
-            0xFFB0C4DE,     // LightSteelBlue
-            0xFFFFFFE0,     // LightYellow
-            0xFF00FF00,     // Lime
-            0xFF32CD32,     // LimeGreen
-            0xFFFAF0E6,     // Linen
-            0xFFFF00FF,     // Magenta
-            0xFF800000,     // Maroon
-            0xFF66CDAA,     // MediumAquamarine
-            0xFF0000CD,     // MediumBlue
-            0xFFBA55D3,     // MediumOrchid
-            0xFF9370DB,     // MediumPurple
-            0xFF3CB371,     // MediumSeaGreen
-            0xFF7B68EE,     // MediumSlateBlue
-            0xFF00FA9A,     // MediumSpringGreen
-            0xFF48D1CC,     // MediumTurquoise
-            0xFFC71585,     // MediumVioletRed
-            0xFF191970,     // MidnightBlue
-            0xFFF5FFFA,     // MintCream
-            0xFFFFE4E1,     // MistyRose
-            0xFFFFE4B5,     // Moccasin
-            0xFFFFDEAD,     // NavajoWhite
-            0xFF000080,     // Navy
-            0xFFFDF5E6,     // OldLace
-            0xFF808000,     // Olive
-            0xFF6B8E23,     // OliveDrab
-            0xFFFFA500,     // Orange
-            0xFFFF4500,     // OrangeRed
-            0xFFDA70D6,     // Orchid
-            0xFFEEE8AA,     // PaleGoldenrod
-            0xFF98FB98,     // PaleGreen
-            0xFFAFEEEE,     // PaleTurquoise
-            0xFFDB7093,     // PaleVioletRed
-            0xFFFFEFD5,     // PapayaWhip
-            0xFFFFDAB9,     // PeachPuff
-            0xFFCD853F,     // Peru
-            0xFFFFC0CB,     // Pink
-            0xFFDDA0DD,     // Plum
-            0xFFB0E0E6,     // PowderBlue
-            0xFF800080,     // Purple
-            0xFFFF0000,     // Red
-            0xFFBC8F8F,     // RosyBrown
-            0xFF4169E1,     // RoyalBlue
-            0xFF8B4513,     // SaddleBrown
-            0xFFFA8072,     // Salmon
-            0xFFF4A460,     // SandyBrown
-            0xFF2E8B57,     // SeaGreen
-            0xFFFFF5EE,     // SeaShell
-            0xFFA0522D,     // Sienna
-            0xFFC0C0C0,     // Silver
-            0xFF87CEEB,     // SkyBlue
-            0xFF6A5ACD,     // SlateBlue
-            0xFF708090,     // SlateGray
-            0xFFFFFAFA,     // Snow
-            0xFF00FF7F,     // SpringGreen
-            0xFF4682B4,     // SteelBlue
-            0xFFD2B48C,     // Tan
-            0xFF008080,     // Teal
-            0xFFD8BFD8,     // Thistle
-            0xFFFF6347,     // Tomato
-            0xFF40E0D0,     // Turquoise
-            0xFFEE82EE,     // Violet
-            0xFFF5DEB3,     // Wheat
-            0xFFFFFFFF,     // White
-            0xFFF5F5F5,     // WhiteSmoke
-            0xFFFFFF00,     // Yellow
-            0xFF9ACD32,     // YellowGreen
+            Unknown,
```

Comment:
should we instead introduce 3 constants?

Reply:
Thanks @safern , yes I did consider that initially but chose the enum route in favor of constants due to easier readability and maybe simpler when using the library? I believe enum compiles to a constant so I went with that currently. Also by using enums we can expand the categories in the future to hold more values other than unknown, system, and web. But you're right, if we use constants directly there will be less casting involved. I will follow your recommendation.

Reply:
I doubt having more categories. If we end up in the need to using more categories we can go back to an enum. 

Yeah, enums are handy cause they compile to a constant, but we are doing a lot of castings for this scenario, so it might just be better to store them as uint constants, or even byte constants given that we don't need 32 bits for the current categories?

Reply:
Sure, sounds good. I will change the enum categories to byte constants. Just a quick question, I had originally made them uint so I could put them in the same uint array:

`public static readonly uint[,] s_colorTable = new uint[knownColorCount, 2]`

but it turned out I still needed to cast the enums. By using byte constants, I will still need to cast to place them inside this table right? Or do you recommend something like an array of structs to hold the category byte constant and color value so we don't need to cast?

Reply:
We could have an array of byte, uint tiple. But let's leave it as you have it now using uint for both values. I'm not sure what would be better. 

@tannergooding insight might help.

Reply:
Thanks, I've updated it with the requested changes, keeping constant uint categories as suggested for now.

Reply:
Hi @safern , anything else I can do to enhance this PR? Thanks for the help.

Reply:
Sorry, I missed the ping and Santi just pinged me offline.

My recommendation is to split this into a `byte[]` (for the kind) and `uint[]` (for the value). The JIT and C# compiler can optimize this scenario and it will save us some space.
The `byte[]` for the kind could even be further compressed if space was a concern, since its basically a bitmap (0 or 1, per entry)

Reply:
Hi @L2, I just spoke with @tannergooding and we think the best approach for this would be to have 2 separate arrays.

One `byte[]` for the color kind and one `uint[]` for the color itself. Having a multidimensional array might affect startup perf and memory perf vs two separate arrays, plus since these are primitives, the JIT will optimize these 2 arrays.

Reply:
Thank you, @tannergooding and @safern . I've updated it with the recommended changes. Looks to have a nice bump in speed with the casting removed and split array.

----------------
dotnet/performance microbenchmarks filtered with System.Drawing.Tests.Perf_Color*:

summary:
better: 3, geomean: 1.480
total diff: 3

No Slower results for the provided threshold = 2% and noise filter = 25ns.

| Faster                                        | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
| --------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Drawing.Tests.Perf_Color.GetBrightness |      1.55 |          1542.74 |           996.36 |         |
| System.Drawing.Tests.Perf_Color.GetSaturation |      1.50 |          1597.70 |          1064.63 |         |
| System.Drawing.Tests.Perf_Color.GetHue        |      1.40 |          1757.32 |          1258.50 |         |